### PR TITLE
add htslib and samtools recipes for testing of develop

### PR DESCRIPTION
**Pull request by dkj**

### CODE REVIEW

File: recipes/htslib/develop/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  name: htslib-pkg
+  version: "{{ version }}"
+
+about:
+  home: https://github.com/samtools/htslib
+  license: MIT
+  summary: C library for high-throughput sequencing data formats.
+
+build:
+  number: 3
```
Comment (kjsanger) [inline recipes/htslib/develop/meta.yaml]: Build number 0?

File: recipes/samtools/develop/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  name: samtools-pkg
+  version: "{{ version }}"
+
+about:
+  home: https://github.com/samtools/samtools
+  license: MIT
+  summary: C library for high-throughput sequencing data formats.
+
+build:
+  number: 1
```
Comment (kjsanger) [inline recipes/samtools/develop/meta.yaml]: Build number 0?

File: recipes/htslib/develop/meta.yaml (hunk 1 above)
Comment (dkj) [reply recipes/htslib/develop/meta.yaml]: I should have reset when first creating a package with this version - but now people may be testing with the test binary packages I'd rather not drop it to 0 until the version string changes again.

File: recipes/samtools/develop/meta.yaml (hunk 2 above)
Comment (dkj) [reply recipes/samtools/develop/meta.yaml]: I think I had a 1-based vs 0-based moment - but now people may be testing with the test binary packages I'd rather not drop it to 0 until the version string changes again.

**Pull request merged**